src/Temporalio/Client/TemporalClient.Workflow.cs

```diff
@@ -289,8 +289,18 @@ public override async Task<TResult> QueryWorkflowAsync<TResult>(QueryWorkflowInp
                 UpdateWorkflowExecutionResponse resp;
                 do
                 {
-                    resp = await Client.Connection.WorkflowService.UpdateWorkflowExecutionAsync(
-                        req, DefaultRetryOptions(input.Options.Rpc)).ConfigureAwait(false);
+                    try
+                    {
+                        resp = await Client.Connection.WorkflowService.UpdateWorkflowExecutionAsync(
+                            req, DefaultRetryOptions(input.Options.Rpc)).ConfigureAwait(false);
+                    }
+                    catch (Exception e) when (e is OperationCanceledException ||
+                        (e is RpcException rpcErr && (
+                            rpcErr.Code == RpcException.StatusCode.DeadlineExceeded ||
+                            rpcErr.Code == RpcException.StatusCode.Cancelled)))
+                    {
+                        throw new WorkflowUpdateRpcTimeoutOrCanceledException(e);
```

Comment:
Do we not want to keep retrying the start in these cases? Or, I guess right now that doesn't matter since there's no explicit overall "update timeout", only the rpc timeout?

Reply:
No, these are usually caused by user-provided timeout or user-initiated cancel. Right, there is no update timeout yet which would be an application level concern and would be considered an update failure as opposed to an RPC failure.